tests/ceph_ansible/test_ansible_upgrade.py

```diff
@@ -182,11 +182,25 @@ def compare_ceph_versions(pre_upgrade_versions, post_upgrade_versions):
     for name, version in pre_upgrade_versions.items():
         # skipping node-exporter version check as it not supported
         if name != "node-exporter":
-            if "installer" not in name and post_upgrade_versions[name] == version:
-                log.error("Pre upgrade version matches post upgrade version")
-                log.error("{}: {} matches".format(name, version))
-                return 1
-    return 0
+            try:
+                if "installer" not in name and post_upgrade_versions[name] == version:
+                    log.error("Pre upgrade version matches post upgrade version")
+                    log.error("{}: {} matches".format(name, version))
+                    return 1
+            except KeyError:
+                # for handling rgw conatiner names during 3.x 'some-rgw' but in 4.x 'some-rgw-rgw0'
+                for key in post_upgrade_versions:
+                    if any(key.startswith(name)):
```

Comment:
Can we do `if key.startswith(name):`?

Reply:
Yes `any` is not needed here this I can correct it,
Thanks @psathyan for the review.